lib/jxl/enc_patch_dictionary.cc

```diff
@@ -752,63 +749,76 @@ void FindBestPatchDictionary(const Image3F& opsin,
   }
 
   CompressParams cparams = state->cparams;
-  cparams.resampling = 1;
-  cparams.ec_resampling = 1;
   // Recursive application of patches could create very weird issues.
   cparams.patches = Override::kOff;
+  // TODO(veluca): possibly change heuristics here.
+  if (!cparams.modular_mode) {
```

Comment:
Can you split this fix in a separate CL?

Reply:
makes sense, I'll split the enc_patch_dictionary refactoring/fix to a separate pull request (and check a bit better what its effect is)

Reply:
PTAL at https://github.com/libjxl/libjxl/pull/473. I will rebase this one once that one lands.